### PR TITLE
AppDomain safety

### PR DESCRIPTION
**Pull request by BrennanConroy**
Related: https://github.com/aspnet/Caching/issues/203

@rynowak

### CODE REVIEW

**Comment by rynowak** [comment]
If it's possible to add a test like https://github.com/aspnet/HttpAbstractions/pull/672/files#diff-4dde99ae301344b1c85a6db8043121ceR48 could you do that?

**Comment by BrennanConroy** [comment]
I could add an `ActionContextAccessor` test

**Comment by BrennanConroy** [comment]
🆙 📅

File: src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/ActionContextAccessor.cs [hunk 1]
```diff
@@ -13,7 +14,7 @@ namespace Microsoft.AspNetCore.Mvc.Infrastructure
     public class ActionContextAccessor : IActionContextAccessor
     {
 #if NET451
-        private static string Key = typeof(ActionContext).FullName;
+        private static string Key = typeof(ActionContext).FullName + AppDomain.CurrentDomain.Id;
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/ActionContextAccessor.cs]: `private static readonly string`

**Comment by rynowak** [comment]
:shipit:

**Pull request merged**